consensus/src/marshal/cache.rs

```diff
@@ -203,6 +234,10 @@ impl<R: Rng + Spawner + Metrics + Clock + GClock + Storage, B: Block, V: Variant
         archive
     }
 
+    pub(crate) fn get_signing_scheme(&mut self, epoch: Epoch) -> Option<S> {
```

Comment:
nit: remove thin wrapper?

Reply:
This is because the provider is currently only stored in the cache. We can remove the wrapper if we also store a clone in the marshal actor.

Reply:
No longer a thin wrapper since we cache the signing scheme.